### DISCUSSION TITLE
Why not embed sanity studio?

### DISCUSSION THREAD

**Discussion by panzacoder** [nuotsu/sanitypress-2023#12] [General]
Hi @nuotsu, I've been playing around with your starter and a few others, and I'm curious why you aren't running sanity from within the NextJS app router, like the [Sanity-created starter](https://github.com/sanity-io/template-nextjs-personal-website).

Thanks for the great work on this template otherwise!

**Comment by nuotsu** [comment]
Thanks for the nice comment! As for not embedding the Studio, although I do find it more organized in terms of the codebase, this had been a deliberate personal choice for the following reason:

When running a local server, saving a Sanity file—a schema file, for instance—would perform a forced reload (not a hot reload) of the Next.js frontend site, which I found quite annoying. I'm just waiting for a possible future fix or improvement from the Sanity team so this experience is a bit better.

I don't think it'd be much of a hassle to move the directory and setup for an embedded Studio so feel free to adjust to fit your needs!

**Comment by mitchuman** [comment]
@panzacoder 
Wanted to let you know, as of the newly published v4.0.0, the studio is now embedded within the Next.js frontend (under the URL `/admin`).

I had played around with the Embedded Studio in a sandbox environment and the issues I encountered initially were resolved, so went along and updated the codebase.
